kolibri/plugins/learn/assets/src/state/actions/recommended.js

```diff
@@ -0,0 +1,280 @@
+  const channelPayload = { channel_id: channelId };
+
+  return ContentNodeResource.getAllContentCollection(channelPayload, { cursor }).fetch();
+}
+
+function showLearnChannel(store, channelId, cursor) {
+  // Special case for when only the page number changes:
+  // Don't set the 'page loading' boolean, to prevent flash and loss of keyboard focus.
+  const state = store.state;
+  if (state.pageName !== PageNames.LEARN_CHANNEL || getCurrentChannelId(state) !== channelId) {
+    store.dispatch('CORE_SET_PAGE_LOADING', true);
+  }
+
+  // session should already be set by app.js
+  // const sessionPromise = SessionResource.getModel('current').fetch();
+  // do we really need to set the channel first?
+  // ConditionalPromise.all([sessionPromise, channelsPromise]).only(
+  const channelsPromise = setChannelInfo(store, channelId);
+  channelsPromise.only(
+    samePageCheckGenerator(store),
+    () => {
+      // IDEA learn page checker
+      if (!getCurrentChannelObject(state)) {
+        router.replace({ name: PageNames.CONTENT_UNAVAILABLE });
+        return;
+      }
+      ConditionalPromise.all([
+        _fetchNextSteps(channelId, state),
+        _fetchPopular(channelId),
+        _fetchResume(channelId, state),
+        _fetchOverview(channelId, cursor),
+      ]).only(
+        samePageCheckGenerator(store), //QUESTION when would this check every be false?
+        ([nextSteps, popular, resume, overview]) => {
+          const currentChannel = getCurrentChannelObject(store.state);
+          const pageState = {
+            // Hard to guarantee this uniqueness on the database side, so
+            // do a uniqBy content_id here, to prevent confusing repeated
+            // content items.
+            nextSteps: uniqBy(nextSteps, 'content_id').map(_contentState),
+            popular: uniqBy(popular, 'content_id').map(_contentState),
+            resume: uniqBy(resume, 'content_id').map(_contentState),
+            overview: overview.map(_contentState),
+            channelId: currentChannel.id,
+            // next: allContentCollection.next,
+            // previous: allContentCollection.previous,
+          };
+          store.dispatch('SET_PAGE_STATE', pageState);
+          store.dispatch('CORE_SET_PAGE_LOADING', false);
+          store.dispatch('CORE_SET_ERROR', null);
+          store.dispatch('SET_PAGE_NAME', PageNames.LEARN_CHANNEL);
+
+          store.dispatch('CORE_SET_TITLE', `Learn - ${currentChannel.title}`);
+        },
+        error => {
+          handleApiError(store, error);
+        }
+      );
+    },
+    error => {
+      handleApiError(store, error);
+    }
+  );
+}
+
+function showPopularPage(store, channelId) {
+  const state = store.state;
+
+  const getPopular = new Promise((resolve, reject) => {
+    const comingFromLearn = state.pageName === PageNames.LEARN_CHANNEL;
+    const channelChanged = getCurrentChannelId(state) !== channelId;
+
+    if (!comingFromLearn || channelChanged) {
```

Comment:
@DXCanas also good to note that our fetches are already cached - you can call them repeatedly, but it will only ever fetch from the server once. The conditional promises are not there to manage client side caching.